src/analytics/src/main/scala/osmesa/analytics/oneoffs/ChangesetORCCreator.scala

```diff
@@ -0,0 +1,175 @@
+        val endSequence = endTime.map(findSequenceFor(_, changesetSource)).getOrElse(getCurrentSequence(changesetSource).sequence)
+
+        val options = Map(
+          Source.BaseURI -> changesetSource.toString,
+          Source.StartSequence -> startSequence.toString,
+          Source.EndSequence -> (endSequence + 1).toString // sequence range is (]; end sequence is exclusive
+        )
+
+        val changesets = spark.read.format(Source.Changesets).options(options).load
+        changesets
+          .repartition(1)
+          .write
+          .orc(outputURI.toString)
+
+        spark.stop()
+      }
+    }
+)
+
+object ChangesetORCUpdaterUtils {
+  implicit val readInstant: Argument[Instant] = new Argument[Instant] {
+    override def read(string: String): ValidatedNel[String, Instant] = {
+      try { Validated.valid(Instant.parse(string)) }
+      catch { case e: Exception => Validated.invalidNel(s"Invalid time: $string (${ e.getMessage })") }
+    }
+
+    override def defaultMetavar: String = "time"
+  }
+
+  // implicit val TimestampFormat : Decoder[Timestamp] = new Decoder[Timestamp] {
+  //   private val sdf = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss XXX")
+
+  //   override def apply(c: HCursor): Decoder.Result[Timestamp] =
+  //     c.get[String]("last_run").map{ str =>
+  //       val splt = str.split(".")
+  //       Timestamp.from(sdf.parse(splt(0) ++ splt(1).split(" ")(1)).toInstant)
+  //     }
+  // }
+
+  private val formatter = DateTimeFormat.forPattern("y-M-d H:m:s.SSSSSSSSS Z")
+
+  private implicit val dateTimeDecoder: Decoder[DateTime] =
+    Decoder.instance(a => a.as[String].map(DateTime.parse(_, formatter)))
+
+  case class Sequence(last_run: DateTime, sequence: Long)
+
+  def getCurrentSequence(baseURI: URI): Sequence = {
+    val response =
+      Http(baseURI.resolve("state.yaml").toString).asString
+
+    val state = yaml.parser
+      .parse(response.body)
+      .leftMap(err => err: Error)
+      .flatMap(_.as[Sequence])
+      .valueOr(throw _)
+
+    state
+  }
+
+  def getSequence(baseURI: URI, sequence: Long): Sequence = {
+    val s = f"${sequence+1}%09d"
+    val path = s"${s.slice(0, 3)}/${s.slice(3, 6)}/${s.slice(6, 9)}.state.txt"
+
+    val response =
+      Http(baseURI.resolve(path).toString).asString
+
+    val state = yaml.parser
+      .parse(response.body)
+      .leftMap(err => err: Error)
+      .flatMap(_.as[Sequence])
+      .valueOr(throw _)
+
+    state
+  }
+
+  def estimateSequenceNumber(modifiedTime: Instant, baseURI: URI): Long = {
+    val current = getCurrentSequence(baseURI)
+    val diffMinutes = (current.last_run.toInstant.getMillis/1000 - modifiedTime.getEpochSecond) / 60
+    current.sequence - diffMinutes
+  }
+
+  def findSequenceFor(modifiedTime: Instant, baseURI: URI): Long = {
+    var guess = estimateSequenceNumber(modifiedTime, baseURI)
+    val target = org.joda.time.Instant.parse(modifiedTime.toString)
```

Comment:
Is Joda-Time necessary here, or does `java.time` have what we need now?

Reply:
Joda time was required to deal with the fractional seconds portion of the date/time strings provided by the state files from overpass.  Java time can deal with milliseconds, which are not the same as fractional seconds.  Or so it seems.  Perhaps testing is required to see if we need to drop this part of the date/time string?

Reply:
Let's just keep using Joda-Time to avoid manipulating the string..  It seems like there should be a way to use `java.time` (with a `DateTimeFormatter`?) but it's certainly not straightforward...

```
scala> import java.time.Instant
import java.time.Instant

scala> Instant.parse("2019-08-06 22:11:02.531677000 +00:00")
java.time.format.DateTimeParseException: Text '2019-08-06 22:11:02.531677000 +00:00' could not be parsed at index 10
  at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2046)
  at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1948)
  at java.base/java.time.Instant.parse(Instant.java:395)
  ... 28 elided
```